src/server/modes/opensource/app.ts

```diff
@@ -71,6 +90,89 @@ function initDataLensApp({
     afterAuth.push(xDlContext(), getCtxMiddleware());
 }
 
+function initZitadel({beforeAuth}: {beforeAuth: AppMiddleware[]}) {
+    if (!zitadelUri) {
+        throw new Error('Missing ZITADEL_URI in env');
+    }
+    if (!clientId) {
+        throw new Error('Missing CLIENT_ID in env');
+    }
+    if (!clientSecret) {
+        throw new Error('Missing CLIENT_SECRET in env');
+    }
+    if (!appHostUri) {
+        throw new Error('Missing APP_HOST_URI in env');
+    }
+    if (!zitadelCookieSecret) {
+        throw new Error('Missing ZITADEL_COOKIE_SECRET in env');
+    }
+
+    passport.use(
+        new OpenIDConnectStrategy(
+            {
+                issuer: zitadelUri,
+                authorizationURL: `${zitadelUri}/oauth/v2/authorize`,
+                tokenURL: `${zitadelUri}/oauth/v2/token`,
+                userInfoURL: `${zitadelUri}/oidc/v1/userinfo`,
+                clientID: clientId,
+                clientSecret: clientSecret,
+                callbackURL: `${appHostUri}/api/auth/callback`,
+                scope: [
+                    'openid',
+                    'profile',
+                    'email',
+                    'offline_access',
+                    'urn:zitadel:iam:org:project:id:zitadel:aud',
+                ],
+                prompt: '1',
+            },
+            (
+                _issuer: string,
+                uiProfile: object,
+                _idProfile: object,
+                _context: object,
+                _idToken: string | object,
+                accessToken: string | object,
+                refreshToken: string,
+                _params: unknown,
+                done: VerifyCallback,
+            ) => {
+                if (typeof accessToken !== 'string') {
+                    throw new Error('Incorrect type of accessToken');
+                }
+
+                const {emails, username, displayName} = uiProfile as any;
+
+                const email = emails?.length > 0 ? emails[0] : undefined;
+                const login = username;
+                const userName = displayName;
+
+                return done(null, {accessToken, refreshToken, login, email, userName});
+            },
+        ),
+    );
+
+    passport.serializeUser((user: Express.User | null | undefined, done) => {
+        process.nextTick(() => done(null, user));
+    });
+
+    passport.deserializeUser(function (user: Express.User | null | undefined, done): void {
+        process.nextTick(() => done(null, user));
+    });
```

Comment:
It looks a little magical. Why do you need this?

Reply:
It reqires passport js to maintain session: https://github.com/jaredhanson/passport

Reply:
Yep, but why we need to use `process.nextTick`

Reply:
Yea, u may be right, let's turn back to classic boilerplate)